<?xml version="1.0"?><!--
=================================================================================== --><!-- |    This document was autogenerated by xacro from
/home/main/johnny-os/install/xarm_description/share/xarm_description/urdf/xarm_device.urdf.xacro | --><!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | --><!--
=================================================================================== -->
<robot name="UF_ROBOT"><!-- Insert at the beginning of xarm_device_macro.xacro -->
    <material name="Black">
        <color rgba="0.0 0.0 0.0 1.0" />
    </material>
    <material name="Red">
        <color rgba="0.8 0.0 0.0 1.0" />
    </material>
    <material name="White">
        <color rgba="1.0 1.0 1.0 1.0" />
    </material>
    <material name="Silver">
        <color rgba="0.753 0.753 0.753 1.0" />
    </material>
    <link name="world" />
    <joint name="world_joint" type="fixed">
        <parent link="world" />
        <child link="link_base" />
        <origin rpy="0 0 0" xyz="0 0 0" />
    </joint>
    <ros2_control name="ecosim_system" type="system">
        <hardware>
            <plugin>topic_based_ros2_control/TopicBasedSystem</plugin>
            <param name="joint_commands_topic">/ecosim/joint_commands</param>
            <param name="joint_states_topic">/ecosim/joint_states</param>
            <param name="sum_wrapped_joint_states">true</param>
        </hardware>
        <joint name="joint1">
            <command_interface name="position">
                <param name="min">-3.1101767270538954</param>
                <param name="max">3.1101767270538954</param>
            </command_interface>
            <command_interface name="velocity">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" /><!-- <state_interface name="effort"/> -->
        </joint>
        <joint name="joint2">
            <command_interface name="position">
                <param name="min">-2.059</param>
                <param name="max">2.0944</param>
            </command_interface>
            <command_interface name="velocity">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" /><!-- <state_interface name="effort"/> -->
        </joint>
        <joint name="joint3">
            <command_interface name="position">
                <param name="min">-3.1101767270538954</param>
                <param name="max">0.19198</param>
            </command_interface>
            <command_interface name="velocity">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" /><!-- <state_interface name="effort"/> -->
        </joint>
        <joint name="joint4">
            <command_interface name="position">
                <param name="min">-3.1101767270538954</param>
                <param name="max">3.1101767270538954</param>
            </command_interface>
            <command_interface name="velocity">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" /><!-- <state_interface name="effort"/> -->
        </joint>
        <joint name="joint5">
            <command_interface name="position">
                <param name="min">-1.69297</param>
                <param name="max">3.1101767270538954</param>
            </command_interface>
            <command_interface name="velocity">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" /><!-- <state_interface name="effort"/> -->
        </joint>
        <joint name="joint6">
            <command_interface name="position">
                <param name="min">-3.1101767270538954</param>
                <param name="max">3.1101767270538954</param>
            </command_interface>
            <command_interface name="velocity">
                <param name="min">-3.14</param>
                <param name="max">3.14</param>
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" /><!-- <state_interface name="effort"/> -->
        </joint>
    </ros2_control>
    <link name="link_base">
        <inertial>
            <origin rpy="0 0 0" xyz="0.0 0.0 0.09103" />
            <mass value="2.7" />
            <inertia ixx="0.00494875" ixy="-3.5E-06" ixz="1.25E-05" iyy="0.00494174" iyz="1.67E-06"
                izz="0.002219" />
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link_base.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <material name="White" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link_base.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
        </collision>
    </link>
    <link name="link1">
        <inertial>
            <origin rpy="0 0 0" xyz="0.00022 0.02951 -0.0124" />
            <mass value="2.3814" />
            <inertia ixx="0.0058562" ixy="-1.79e-05" ixz="3.55e-06" iyy="0.0050316"
                iyz="-0.000888336" izz="0.003536652" />
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link1.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <material name="White" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link1.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
        </collision>
    </link>
    <joint name="joint1" type="revolute">
        <parent link="link_base" />
        <child link="link1" />
        <origin rpy="0 0 0" xyz="0 0 0.267" />
        <axis xyz="0 0 1" />
        <limit effort="50.0" lower="-3.1101767270538954" upper="3.1101767270538954" velocity="3.14" />
        <dynamics damping="1.0" friction="1.0" />
    </joint>
    <link name="link2">
        <inertial>
            <origin rpy="0 0 0" xyz="0.03881 -0.22783 0.03496" />
            <mass value="2.2675" />
            <inertia ixx="0.028315776" ixy="0.005" ixz="0.00066546" iyy="0.0058" iyz="-0.0045741"
                izz="0.0273447" />
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link2.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <material name="White" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link2.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
        </collision>
    </link>
    <joint name="joint2" type="revolute">
        <parent link="link1" />
        <child link="link2" />
        <origin rpy="-1.5708 0 0" xyz="0 0 0" />
        <axis xyz="0 0 1" />
        <limit effort="50.0" lower="-2.059" upper="2.0944" velocity="3.14" />
        <dynamics damping="1.0" friction="1.0" />
    </joint>
    <link name="link3">
        <inertial>
            <origin rpy="0 0 0" xyz="0.07041 0.11631 0.0107" />
            <mass value="1.875" />
            <inertia ixx="0.0063483" ixy="-0.0015397" ixz="0.00096858" iyy="0.00379758"
                iyz="0.00186567" izz="0.00595768" />
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link3.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <material name="White" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link3.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
        </collision>
    </link>
    <joint name="joint3" type="revolute">
        <parent link="link2" />
        <child link="link3" />
        <origin rpy="0 0 0" xyz="0.0535 -0.2845 0" />
        <axis xyz="0 0 1" />
        <limit effort="32.0" lower="-3.1101767270538954" upper="0.19198" velocity="3.14" />
        <dynamics damping="1.0" friction="1.0" />
    </joint>
    <link name="link4">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.00018 0.01798 -0.02291" />
            <mass value="1.3192" />
            <inertia ixx="0.004896" ixy="-6.925e-06" ixz="-1.418e-05" iyy="0.00445694"
                iyz="-0.00023186" izz="0.00134332" />
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link4.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <material name="White" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link4.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
        </collision>
    </link>
    <joint name="joint4" type="revolute">
        <parent link="link3" />
        <child link="link4" />
        <origin rpy="-1.5708 0 0" xyz="0.0775 0.3425 0" />
        <axis xyz="0 0 1" />
        <limit effort="32.0" lower="-3.1101767270538954" upper="3.1101767270538954" velocity="3.14" />
        <dynamics damping="1.0" friction="1.0" />
    </joint>
    <link name="link5">
        <inertial>
            <origin rpy="0 0 0" xyz="0.0651 0.03096 0.00315" />
            <mass value="1.33854" />
            <inertia ixx="0.00146378" ixy="-0.000450624" ixz="0.000284306" iyy="0.00184192"
                iyz="0.000130866" izz="0.002333524" />
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link5.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <material name="White" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link5.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
        </collision>
    </link>
    <joint name="joint5" type="revolute">
        <parent link="link4" />
        <child link="link5" />
        <origin rpy="1.5708 0 0" xyz="0 0 0" />
        <axis xyz="0 0 1" />
        <limit effort="32.0" lower="-1.69297" upper="3.1101767270538954" velocity="3.14" />
        <dynamics damping="1.0" friction="1.0" />
    </joint>
    <link name="link6">
        <inertial>
            <origin rpy="0 0 0" xyz="0 -0.00677 -0.01098" />
            <mass value="0.17" />
            <inertia ixx="9.3e-05" ixy="-0.0" ixz="-0.0" iyy="5.87e-05" iyz="-3.6e-06"
                izz="0.000132" />
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://xarm_description/meshes/xarm6/visual/link6.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <material name="Silver" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://xarm_description/meshes/end_tool/collision/end_tool.stl" />
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0" />
        </collision>
    </link>
    <joint name="joint6" type="revolute">
        <parent link="link5" />
        <child link="link6" />
        <origin rpy="-1.5708 0 0" xyz="0.076 0.097 0" />
        <axis xyz="0 0 1" />
        <limit effort="20.0" lower="-3.1101767270538954" upper="3.1101767270538954" velocity="3.14" />
        <dynamics damping="1.0" friction="1.0" />
    </joint><!-- <link name="${prefix}link_eef"
    />\n
    <joint name="${prefix}joint_eef" type="fixed">\n      <origin xyz="0 0 0" rpy="0 0 0" />\n
    <parent link="${prefix}link6" />\n      <child link="${prefix}link_eef" />\n    </joint> -->
    <transmission name="tran1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>100</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="tran2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>100</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="tran3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>100</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="tran4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint4">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor4">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>100</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="tran5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint5">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor5">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>100</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="tran6">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint6">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor6">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>100</mechanicalReduction>
        </actuator>
    </transmission>
    <gazebo reference="link_base">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link1">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link2">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link3">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link4">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link5">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="link6">
        <selfCollide>true</selfCollide>
    </gazebo>
    <link name="link_eef" />
    <joint name="joint_eef" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0" />
        <parent link="link6" />
        <child link="link_eef" />
    </joint>
    <gazebo>
        <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
            <robot_sim_type>gazebo_ros2_control/GazeboSystem</robot_sim_type>
            <parameters>
                /home/main/johnny-os/install/xarm_controller/share/xarm_controller/config/xarm6_controllers.yaml</parameters>
        </plugin>
    </gazebo><!--
    Insert at the end of xarm_device_macro.xacro -->
</robot>
